karaf/s2i/run

```diff
@@ -3,6 +3,19 @@
 # Global S2I variable setup
 source `dirname "$0"`/s2i-setup
 
+# Setup Jolokia to accept basic auth, using a randomly generated password that is stored
+# in the container in the ${DEPLOYMENTS_DIR}/jolokia.pw file.
+AB_JOLOKIA_USER="jolokia"
+if [ -f "${DEPLOYMENTS_DIR}/jolokia.pw" ] ; then
+	AB_JOLOKIA_PASSWORD=`cat "${DEPLOYMENTS_DIR}/jolokia.pw"`
+else
+	AB_JOLOKIA_PASSWORD=`tr -cd '[:alnum:]' < /dev/urandom | fold -w30 | head -n1`
+	touch "${DEPLOYMENTS_DIR}/jolokia.pw"
+	chmod 600 "${DEPLOYMENTS_DIR}/jolokia.pw"
+	echo $AB_JOLOKIA_PASSWORD > "${DEPLOYMENTS_DIR}/jolokia.pw"
+fi
+export AB_JOLOKIA_USER
```

Comment:
> Your right, we need to update it so that it uses setWantClientAuth() instead of setNeedClientAuth(). I'll send in a follow up PR for that.

Yes, that could work when we also add some extra verification into the ClientCertAuthenticator. See https://github.com/rhuss/jolokia/pull/275 for some thoughts on this.